index.js

```diff
@@ -22,8 +22,10 @@ module.exports = async function (fastify, opts) {
 
   fastify.register(From, fromOpts)
 
-  fastify.addContentTypeParser('application/json', bodyParser)
-  fastify.addContentTypeParser('*', bodyParser)
+  if (opts.proxyPayloads !== false) {
+    fastify.addContentTypeParser('application/json', bodyParser)
+    fastify.addContentTypeParser('*', bodyParser)
+  }
```

Comment:
@amiteshtoharya4 this is the bit we are discussing. Without any registered content type handlers, you are at best relying on Fastify's default implementation (in particularly, `application/json` will be parsed by Fastify) and at worst completely dropping the incoming payload. The default handler in this plugin is a pass-through handler. It removes Fastify's default handling and passes the body through to the proxied system for _all_ content types. So, by setting `opts.proxyPayloads = false` you are responsible for knowing these things and handling them appropriately.

Reply:
got it.... will add an example of payload handling for some content-type